app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -210,7 +217,116 @@ private void processFromLoginLaunch() {
             if (CommCareHomeActivity.isDemoUser()) {
                 showDemoModeWarning();
             }
+
+            pinModeDecisionLogic();
+        }
+    }
+
+    // See if we should launch the pin choice dialog
+    private void pinModeDecisionLogic() {
+
+        LoginActivity.LoginMode loginMode = LoginActivity.LoginMode.fromString(
+                getIntent().getStringExtra(LoginActivity.LOGIN_MODE));
+
+        if (loginMode == LoginActivity.LoginMode.PRIMED) {
+            launchPinCreateScreen(loginMode);
```

Comment:
If this is going to hit another activity it's probably a good idea to return afterwards to make sure the workflow doesn't get hijacked.